source/code/plugins/agent_telemetry_script.rb

```diff
@@ -177,6 +177,17 @@ def self.clear
       @@qos_events.clear
     end
 
+    def self.push_back_qos_event(source, event)
+      if @@qos_events.has_key?(source)
+        if @@qos_events[source].size >= QOS_EVENTS_LIMIT
+          @@qos_events[source].shift # remove oldest qos event to cap memory use
```

Comment:
why is this separate fn needed?

Reply:
I have call push_back_qos_event to push QoS event passed from background jobs.